File: nova-app/src-tauri/src/commands/file_system.rs
use authenticated_command::authenticated_command;
use nova_fs::file_system::FileSystem;

#[authenticated_command]
pub async fn read_file_to_string(file: String) -> Result<String, String> {
    FileSystem::read_async(file).await
}

#[authenticated_command]
pub fn join(parts: Vec<String>) -> Result<String, String> {
    Ok(FileSystem::join(parts))
}

#[tauri::command]
pub async fn create_dir(dir: String) -> Result<bool, String> {
    Ok(FileSystem::create_dir_async(dir).await)
}

#[authenticated_command]
pub async fn create_dir_recursive(dir: String) -> Result<bool, String> {
    Ok(FileSystem::create_dir_recursive_async(dir).await)
}

#[authenticated_command]
pub async fn remove_dir(dir: String) -> Result<bool, String> {
    Ok(FileSystem::remove_dir_async(dir).await)
}

#[authenticated_command]
pub async fn remove_dir_recursive(dir: String) -> Result<bool, String> {
    Ok(FileSystem::remove_dir_recursive_async(dir).await)
}

#[authenticated_command]
pub async fn remove_file(file: String) -> Result<bool, String> {
    Ok(FileSystem::remove_file_async(file).await)
}

#[authenticated_command]
pub async fn rename_path(from: String, to: String) -> Result<bool, String> {
    Ok(FileSystem::rename_async(from, to).await)
}

#[authenticated_command]
pub async fn path_exists(path: String) -> Result<bool, String> {
    Ok(FileSystem::exists_async(path).await)
}

#[authenticated_command]
pub async fn write_file(path: String, contents: String) -> Result<bool, String> {
    Ok(FileSystem::write_async(path, contents).await)
}

#[authenticated_command]
pub async fn is_empty(path: String) -> Result<bool, String> {
    Ok(FileSystem::is_empty_async(path).await)
}

File: nova-web/src/lib/FileSystem.ts
import {NovaFileSystemApi} from "../nova_api/FileSystemApi.ts";
import {type Result} from "./Result.ts";

export class FileSystem {
    public static async read(path: string): Promise<Result<string>> {
        return NovaFileSystemApi.read(path);
    }

    public static async write(path: string, contents: string): Promise<boolean> {
        return NovaFileSystemApi.write(path, contents);
    }

    public static async createDir(path: string): Promise<boolean> {
        return NovaFileSystemApi.createDir(path);
    }

    public static async createDirRecursive(path: string): Promise<boolean> {
        return NovaFileSystemApi.createDirRecursive(path);
    }

    public static async removeDir(dir: string): Promise<boolean> {
        return NovaFileSystemApi.removeDir(dir);
    }

    public static async removeDirRecursive(dir: string): Promise<boolean> {
        return NovaFileSystemApi.removeDirRecursive(dir);
    }

    public static async removeFile(file: string): Promise<boolean> {
        return NovaFileSystemApi.removeFile(file);
    }

    public static async rename(from: string, to: string): Promise<boolean> {
        return NovaFileSystemApi.rename(from, to);
    }

    public static async exist(path: string): Promise<boolean> {
        return NovaFileSystemApi.exist(path);
    }

    public static async isEmpty(path: string): Promise<boolean> {
        return NovaFileSystemApi.isEmpty(path);
    }

    public static async join(parts: string[]): Promise<string> {
        return NovaFileSystemApi.join(parts);
    }
}